devito/types/basic.py

```diff
@@ -426,8 +437,114 @@ def __dtype_setup__(cls, **kwargs):
         return kwargs.get('dtype', np.float32)
 
 
-class AbstractFunction(sympy.Function, Basic, Cached, Pickable, Evaluable):
+class AbstractTensor(sympy.ImmutableDenseMatrix, Basic, Cached, Pickable, Evaluable):
```

Comment:
Should `AbstractTensor` be referenced in `class Basic` docstring or you feel having `AbstractObject` is sufficient?

Reply:
Indeedd forgot about this one thanks.

Reply:
fixed